shell/server/pre-meteor.js

```diff
@@ -43,6 +43,12 @@ function isSandstormShell(hostname) {
   return (hostname === HOSTNAME || (DDP_HOSTNAME && hostname === DDP_HOSTNAME));
 }
 
+function hasPseudoTLD(hostname) {
+  // Returns true if this hostname ends in one of the major pseudo-TLDs.
+  let tld = hostname.split(".").pop();
+  return (tld === ("onion" || "i2p" || "gnu"));
```

Comment:
Oh, interesting, I see now.

Your way is almost fine, but I think the short-circuit evaluation is going to be bad news; I think you're `||`-ing the TLD names, not `||`-ing the result of comparing your desired string against the TLD names.

https://epeli.github.io/underscore.string/#endswith-string-ends-position-gt-boolean supposedly exists.

I don't know if you're familiar with the Sandstorm dev flow, so here's how I am checking how we'd use that:

```
$ cd ~/projects/sandstorm/shell
$ ./run-dev.sh
# lots of printouts while stuff compiles
# slow last line while it fuzz tests capnp
I couldn't find an initscript for Sandstorm. Please pass the directory
where Sandstorm is installed as an argument to this script.
$ ➜  ./run-dev.sh /opt/sandstorm
Please shut down your Sandstorm front-end:
  sudo /opt/sandstorm/sandstorm stop-fe
$ sudo /opt/sandstorm/sandstorm stop-fe                
$ ./run-dev.sh /opt/sandstorm
# more printout
=> App running at: http://local.sandstorm.io:6080
```

In a different terminal:

```
$ cd ~/projects/sandstorm/shell
$ meteor shell
# some printout
> _.endsWith('a')
TypeError: Object function (obj) {
    if (obj instanceof _) return obj;
    if (!(this instanceof _)) return new _(obj);
    this._wrapped = obj;
  } has no method 'endsWith'
    at repl:1:4
    at /home/paulproteus/projects/sandstorm/shell/.meteor/local/build/programs/server/shell-server.js:243:23
```

OK, so I guess we have to do something like

```
    meteor add underscorestring:underscore.string
```

per https://github.com/epeli/underscore.string . That seems like a lot of work for one `endsWith()` function, but it could be a reasonable idea.